### ISSUE TITLE
Weblate removes _zero postfixed keys from translation files that were touched

### ISSUE THREAD

**Issue by zakharchenkoAndrii** [WeblateOrg/weblate#9358]
### Describe the issue

Hi! The issue already was described [here](https://github.com/WeblateOrg/weblate/issues/8794#issuecomment-1490777508)  as a comment

Preconditions:
* i18next 22.3.0
* weblate server version is 4.17-2

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

Have keys with `_zero` postfix

1. Go to weblate
2. Change any translation entry
3. Observe that along with original intended change, all keys with _zero has been removed

### Expected behavior

Keys are not removed

### How do you run Weblate?

Docker container

### Weblate versions

4.17-2

**Comment by nijel** [comment]
The file handling is done by https://github.com/translate/translate, so if something needs to be changed, it has to be done there.

How does this actually work in i18next? The current assumption is that it follows CLDR plural rules like anybody else. So, these `_zero` suffixes would not be used if there are no matching CLDR plural for given language.

**Comment by zakharchenkoAndrii** [comment]
I mean... even though we are not doing it by standard, why it removes keys completely? 

why not just keeping keys as it is, and treat them as separate translation values on the dashboard...

**Comment by nijel** [comment]
Because it was the easiest way to implement :-). Doing what you describe involves some corner cases. Possible solutions:

* Introducing different plural rules which would include zero in all cases. Weblate already does this for Stringsdict where it is mandated by the format. But this makes the decision on the translation, not on the string level.
* Handling _zero suffixes as separate strings when present in the source language. That would require additional logic to skip such strings for languages which have zero rule included.

**Comment by zakharchenkoAndrii** [comment]
The first option sounds more reasonable 🤔 . The Second one kind of ambiguous and can confuse users

**Comment by nijel** [comment]
A similar situation is with Rails Internationalization (I18n) API, it can have optional zero plural as well, see https://guides.rubyonrails.org/i18n.html#pluralization.

**Comment by nijel** [comment]
This feature of i18next was introduced in https://github.com/i18next/i18next/releases/tag/v21.6.0

**Comment by zakharchenkoAndrii** [comment]
> This feature of i18next was introduced in https://github.com/i18next/i18next/releases/tag/v21.6.0

yes, and we are using this feature for more natural localization. That's, probably, why we have this problem. (our i18next version is 22.3.0)

**Comment by nijel** [comment]
I think we will go the same way as with Apple StringsDict and always include zero form for these two formats. It just needs coordinated changes in Weblate and translate-toolkit.

**Comment by ahus1** [comment]
I'm new to Weblate and I stumbled across this issue when handling our i18next v4 files. For one of the files I've chosen by accident the v3 version, which uses numeric suffixes instead. Once I did that, Weblate treated them as separate keys. 

So the workaround for me would be to use i18next v3 (or plain nested JSON) as long as my languages have the same suffix? Do I happen to miss some problem lurking here?

**Comment by yarons** [comment]
Is there an open PR in translate-tookkit we can follow?

**Comment by MaximeDametis** [comment]
This feature would be really helpful for my application
